src/ui/server/controllers/insurance/business/company-details/index.test.ts

```diff
@@ -271,24 +264,150 @@ describe('controllers/insurance/business/companies-details', () => {
+      const errorMessageCompanyHouseIssue = EXPORTER_BUSINESS_ERROR[COMPANY_HOUSE.INPUT].NOT_FOUND;
+
+      let validationErrors = generateValidationErrors(COMPANY_HOUSE.INPUT, errorMessageCompanyHouseIssue, {});
+      validationErrors = generateValidationErrors(TRADING_NAME, errorMessageTradingName, validationErrors);
+
+      expect(res.render).toHaveBeenCalledWith(companyDetailsTemplate, {
+        ...corePageVariables({
+          PAGE_CONTENT_STRINGS: COMPANY_DETAILS,
+          BACK_LINK: req.headers.referer,
+        }),
+        ...pageVariables(mockApplication.referenceNumber),
+        validationErrors,
+        submittedValues,
+      });
+    });
+
+    it('should display validation error for trading anme when there is no trading name and company is found', async () => {
+      req.body = {
+        companiesHouseNumber: '123456',
+      };
+
+      const submittedValues = {
+        [COMPANY_HOUSE.INPUT]: req.body[COMPANY_HOUSE.INPUT],
+        [TRADING_NAME]: sanitiseValue(req.body[TRADING_NAME]),
+      };
+
+      const getCompaniesHouseResponse = jest.fn(() => Promise.resolve(mockCompany));
+      api.keystone.getCompaniesHouseInformation = getCompaniesHouseResponse;
+
+      await post(req, res);
+
+      const errorMessageTradingName = EXPORTER_BUSINESS_ERROR[TRADING_NAME].IS_EMPTY;
+
+      expect(res.render).toHaveBeenCalledWith(companyDetailsTemplate, {
+        ...corePageVariables({
+          PAGE_CONTENT_STRINGS: COMPANY_DETAILS,
+          BACK_LINK: req.headers.referer,
+        }),
+        ...pageVariables(mockApplication.referenceNumber),
+        validationErrors: generateValidationErrors(TRADING_NAME, errorMessageTradingName, {}),
+        submittedValues,
+      });
+    });
+
+    it('should display validation error for companies house when company house error but trading name radio is selected', async () => {
+      req.body = {
+        companiesHouseNumber: '123456',
+        [TRADING_NAME]: 'true',
+      };
+
+      const submittedValues = {
+        [COMPANY_HOUSE.INPUT]: req.body[COMPANY_HOUSE.INPUT],
+        [TRADING_NAME]: sanitiseValue(req.body[TRADING_NAME]),
+      };
+
+      const getCompaniesHouseResponse = jest.fn(() => Promise.resolve({ apiError: true }));
+      api.keystone.getCompaniesHouseInformation = getCompaniesHouseResponse;
+
+      await post(req, res);
+
+      const errorMessage = EXPORTER_BUSINESS_ERROR[COMPANY_HOUSE.INPUT].TECHNICAL_ISSUES;
 
-      const errorMessage = EXPORTER_BUSINESS_ERROR[TRADING_NAME].IS_EMPTY;
       expect(res.render).toHaveBeenCalledWith(companyDetailsTemplate, {
         ...corePageVariables({
           PAGE_CONTENT_STRINGS: COMPANY_DETAILS,
           BACK_LINK: req.headers.referer,
         }),
         ...pageVariables(mockApplication.referenceNumber),
-        validationErrors: generateValidationErrors(TRADING_NAME, errorMessage, {}),
+        validationErrors: generateValidationErrors(COMPANY_HOUSE.INPUT, errorMessage, {}),
```

Comment:
Are we able to split this file up somehow? Nearly 450 lines, maybe a single file for GET and another for POST?

Reply:
split to index.POST.test.js index.test.js